gravitee-am-service/src/test/java/io/gravitee/am/service/impl/PasswordHistoryServiceTest.java

```diff
@@ -0,0 +1,228 @@
+/**
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.am.service.impl;
+
+import io.gravitee.am.identityprovider.api.DefaultUser;
+import io.gravitee.am.model.PasswordHistory;
+import io.gravitee.am.model.PasswordSettings;
+import io.gravitee.am.model.ReferenceType;
+import io.gravitee.am.repository.management.api.PasswordHistoryRepository;
+import io.gravitee.am.service.AuditService;
+import io.gravitee.am.service.authentication.crypto.password.PasswordEncoder;
+import io.gravitee.am.service.exception.PasswordHistoryException;
+import io.gravitee.am.service.reporter.builder.management.UserAuditBuilder;
+import io.reactivex.Completable;
+import io.reactivex.Single;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.mockito.ArgumentCaptor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Stream;
+
+import static io.gravitee.am.model.ReferenceType.DOMAIN;
+import static io.reactivex.Flowable.fromIterable;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.isA;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.verify;
+
+/**
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+@ExtendWith(MockitoExtension.class)
+class PasswordHistoryServiceTest {
```

Comment:
CodeCoverage isn't taking into account for this class, Quality Gate fails

https://sonarcloud.io/component_measures?id=gravitee-io_graviteeio-access-management&pullRequest=2188&metric=new_coverage&view=list

Reply:
Might be because it's package private.

Reply:
Hmm that is weird. I have Sonarlint in intellij as well and it actually recommends default access modifier for Junit 5 tests. I will investigate this.